src/inmanta/plugins.py

```diff
@@ -177,6 +177,18 @@ def get_functions(cls) -> Dict[str, "Type[Plugin]"]:
         """
         return dict(cls.__functions)
 
+    @classmethod
+    def deprecate_function(cls, fnc: Callable, replaced_by: Optional[str] = None) -> None:
+        name = fnc.__name__
```

Comment:
Could you add a test case where you use this new decorator together with a plugin with a custom name? i.e.
```python
@deprecated
@plugin("pluginnamedoesnotmatchfunctioname")
def myfunction():
    pass
```

I expect this will currently not work. I don't know how to fix it yet, let's first see if my hypothesis is right.

Reply:
indeed : `Can not deprecate plugin test_module::get_one as it does not exist` :(

Reply:
wouldn't it be possible to add a field to the dictionary with the "true name" and verify the "true name" is present?

Reply:
@sanderr I didn't want to do a reverse lookup so I've added an extra field. What do you think about that? The naming should still be changed but I has no ideas